### PR TITLE
Improve iozone_windows to support disk test

### PR DESCRIPTION
**Pull request by suqinhuang**
iozone can be run in system disk currently,
add disk support as well, need to create the partition first.

ID: 1303003

Signed-off-by: Suqin Huang <shuang@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
@suqinhuang  it looks we have format disk function in framework, please call it directly

**Comment by spcui** [comment]
agreed with xutian, utils_misc.format_guest_disk() can do it.

File: generic/tests/iozone_windows.py [hunk 1]
```diff
@@ -19,27 +21,45 @@ def run(test, params, env):
...
     results_path = os.path.join(test.resultsdir,
                                 'raw_output_%s' % test.iteration)
     analysisdir = os.path.join(test.resultsdir, 'analysis_%s' % test.iteration)
 
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    if params.get("format_disk", "no") == "yes":
+        error_context.context("Format disk")
```
Comment (CongSmile) [inline generic/tests/iozone_windows.py]: Write the log in logging.info is better.

File: generic/tests/iozone_windows.py [hunk 2]
```diff
@@ -19,27 +21,45 @@ def run(test, params, env):
...
     # Run IOzone and record its results
-    drive_letter = utils_misc.get_winutils_vol(session)
-    c = params["iozone_cmd"] % drive_letter
-    t = int(params.get("iozone_timeout"))
-    logging.info("Running IOzone command on guest, timeout %ss", t)
-    results = session.cmd_output(cmd=c, timeout=t)
+    winutils = utils_misc.get_winutils_vol(session)
+    iozone_cmd = params["iozone_cmd"] % (winutils, disk_letter)
+    error_context.context("Running IOzone command on guest, timeout %ss"
+                          % iozone_timeout)
```
Comment (CongSmile) [inline generic/tests/iozone_windows.py]: Also this line.

**Comment by CongSmile** [comment]
@suqinhuang 

Does this case need enable driver verifier?
I mean for the case only.

**Comment by suqinhuang** [comment]
@CongSmile never do driver verifier before,
and I think we need a  independence case just for iozone testing, which can be called by others.

**Comment by CongSmile** [comment]
@suqinhuang 

Yes, this is a step problem before, has confirmed with the feature owner, need enable driver verifier when the iozone is as a separate case.

File: generic/tests/cfg/iozone_windows.cfg [hunk 3]
```diff
@@ -3,8 +3,11 @@
     only Windows
     type = iozone_windows
     cdrom_cd1 = isos/windows/winutils.iso
-    iozone_cmd = "%s:\Iozone\iozone.exe -azR -r 64k -n 1G -g 4G -M -b iozone.xls -f c:\testfile"
+    disk_letter = "C"
+    iozone_cmd = "WIN_UTILS:\Iozone\iozone.exe -azR -r 64k -n 1G -g 4G -M -b iozone.xls -f ${disk_letter}:\testfile"
     iozone_timeout = 7200
+    need_enable_verifier = yes
```
Comment (CongSmile) [inline generic/tests/cfg/iozone_windows.cfg]: need_clear_verifier?

File: generic/tests/cfg/iozone_windows.cfg [hunk 4]
```diff
@@ -3,8 +3,11 @@
     only Windows
     type = iozone_windows
     cdrom_cd1 = isos/windows/winutils.iso
-    iozone_cmd = "%s:\Iozone\iozone.exe -azR -r 64k -n 1G -g 4G -M -b iozone.xls -f c:\testfile"
+    disk_letter = "C"
```
Comment (vipmike007) [inline generic/tests/cfg/iozone_windows.cfg]: Better use following way to check which disk is the boot disk
diskpart 
list volumes

Comment (suqinhuang) [inline generic/tests/cfg/iozone_windows.cfg]: if test on data disk, we need to assign disk letter to create a partition, or use default one.

**Comment by suqinhuang** [comment]
PASS smp_1.4096m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.Win8.x86_64.1.io-github-autotest-qemu.iozone_windows.aio_threads

File: generic/tests/iozone_windows.py [hunk 5]
```diff
@@ -19,27 +25,70 @@ def run(test, params, env):
...
+                                             output_dir=analysisdir)
+        p.plot_all()
+
+    def get_driver():
+        driver_name = params.get("driver_name", "")
+        drive_format = params.get("drive_format")
+        if not driver_name:
+            if "scsi" in drive_format:
+                driver_name = "vioscsi"
+            elif "virtio" in drive_format:
```
Comment (suqinhuang) [inline generic/tests/iozone_windows.py]: 10:48:11 test             L0209 DEBUG|     drive_format = virtio

**Comment by spcui** [comment]
LGTM,
Acked-by: scui@redhat.com

File: generic/tests/iozone_windows.py [hunk 6]
```diff
@@ -19,27 +25,70 @@ def run(test, params, env):
...
+    if driver_name:
+        if params.get("need_enable_verifier", "no") == "yes":
+            error_context.context("Enable %s driver verifier" % driver_name,
+                                  logging.info)
+            try:
+                session = utils_test.qemu.setup_win_driver_verifier(
+                          session, driver_name, vm, timeout)
+                funcatexit.register(env, params.get("type"),
+                                    utils_test.qemu.clear_win_driver_verifier(
+                                    session, vm, timeout))
```
Comment (CongSmile) [inline generic/tests/iozone_windows.py]: @suqinhuang 

A little confused why clear the verifier here?

Comment (xutian) [inline generic/tests/iozone_windows.py]: Not call it here, it will to call in the end of test.

File: generic/tests/iozone_windows.py [hunk 7]
```diff
@@ -19,27 +25,70 @@ def run(test, params, env):
...
+        a.analyze()
+        p = postprocess_iozone.IOzonePlotter(results_file=results_path,
+                                             output_dir=analysisdir)
+        p.plot_all()
+
+    def get_driver():
+        driver_name = params.get("driver_name", "")
+        drive_format = params.get("drive_format")
+        if not driver_name:
+            if "scsi" in drive_format:
```
Comment (CongSmile) [inline generic/tests/iozone_windows.py]: How about spapr_vscsi for ppc?

Comment (suqinhuang) [inline generic/tests/iozone_windows.py]: @CongSmile Windows is not supported in ppc

File: generic/tests/iozone_windows.py [hunk 8]
```diff
@@ -19,27 +25,70 @@ def run(test, params, env):
...
+    winutils = utils_misc.get_winutils_vol(session)
+    cmd = params["iozone_cmd"]
+    iozone_cmd = re.sub("WIN_UTILS", winutils, cmd)
+    error_context.context("Running IOzone command on guest, timeout %ss"
+                          % iozone_timeout, logging.info)
+
+    status, results = session.cmd_status_output(cmd=iozone_cmd,
+                                                timeout=iozone_timeout)
+    error_context.context("Write results to %s" % results_path, logging.info)
+    if status != 0:
```
Comment (CongSmile) [inline generic/tests/iozone_windows.py]: if status:

File: generic/tests/iozone_windows.py [hunk 9]
```diff
@@ -19,27 +25,70 @@ def run(test, params, env):
     :param params: Dictionary with the test parameters
     :param env: Dictionary with test environment.
     """
-    vm = env.get_vm(params["main_vm"])
-    vm.verify_alive()
+
+    def post_result(results_path, analysisdir):
```
Comment (CongSmile) [inline generic/tests/iozone_windows.py]: Please add the doc for the func()

File: generic/tests/iozone_windows.py [hunk 10]
```diff
@@ -19,27 +25,70 @@ def run(test, params, env):
...
+        a = postprocess_iozone.IOzoneAnalyzer(list_files=[results_path],
+                                              output_dir=analysisdir)
+        a.analyze()
+        p = postprocess_iozone.IOzonePlotter(results_file=results_path,
+                                             output_dir=analysisdir)
+        p.plot_all()
+
+    def get_driver():
```
Comment (CongSmile) [inline generic/tests/iozone_windows.py]: Please add doc for the func().

Others are fine for me.

File: generic/tests/iozone_windows.py (hunk 8 above)
Comment (suqinhuang) [inline generic/tests/iozone_windows.py]: need to raise error when status !=0

**Comment by CongSmile** [comment]
Looks good.
Acked-by: Cong Li <coli@redhat.com>

**Comment by suqinhuang** [comment]
Thanks @spcui @CongSmile @xutian merged

File: generic/tests/iozone_windows.py (hunk 5 above)
Comment (xutian) [inline generic/tests/iozone_windows.py]: @suqinhuang 
below update looks more clear and clean.

if "virtio" in driver_format:
     if "scsi" in driver_format:
           return "vioscsi"
     else:
           return  "viostor"
return None

**Pull request merged**